interactivity/configuring-ai-powered-insights.md

```diff
@@ -0,0 +1,428 @@
+
+> RAG is available only in .NET and .NET Standard. Therefore, the options that are listed below are not supported in .NET Framework configurations.
+
+If needed, you can disable this algorithm by setting `allowRAG` to `false`.
+
+You can also configure the RAG behavior through the `ragSettings` option:
+- `modelMaxInputTokenLimit`&mdash;Limits the maximum input tokens the AI model can process in a single request. The default value is `15000`.
+- `maxNumberOfEmbeddingsSent`&mdash;Limits how many embeddings (chunks of retrieved content) are sent to the model in a single request. The default value is `15`.
+- `maxTokenSizeOfSingleEmbedding`&mdash;Limits token size of each individual embedding, which prevents large chunks from dominating the prompt. The default value is `0` (no limit).
+- `tokenizationEncoding`&mdash;Specifies tokenization scheme used to estimate the tokens usage before sending the request to the LLM model. By default, the encoding is determined automatically based on the specified model, which is recommended to ensure accurate token counting. Incorrect encoding may lead to miscalculations in token limits, causing either premature truncation of context or exceeding the model’s input capacity.
+- `splitTables`&mdash;Indicates whether tables should be split during Retrieval-Augmented Generation (RAG) processing. When the splitting is allowed, only the relevant table cells will be taken into account, significantly reducing the number of tokens. The default value is `true`.
+
+Below is an example that takes advantage of the table splitting and automatic encoding inference, but reduces the token limits:
+
+````JSON
+"telerikReporting": {
+	"AIClient": {
+        // ...base configuration...
+		"requireConsent": false,
+		"allowCustomPrompts": false,
+		"predefinedPrompts": [
+			{ "text": "Generate an executive summary of this report." },
+			{ "text": "Translate the document into German." }
+		],
+		"ragSettings": {
+			"modelMaxInputTokenLimit": 12000,
+			"maxNumberOfEmbeddingsSent": 10,
+			"maxTokenSizeOfSingleEmbedding": 2000
+		}
+	}
+}
+````
+
+For a complete reference of all available `AIClient` options, check the article [AIClient Element Overview]({%slug telerikreporting/aiclient-element%}).
+
+## Overriding ReportsControllerBase Methods
+
+While the [declarative configuration](#configuring-the-report-engine) handles most common scenarios, some advanced use cases require programmatic customization. You can achieve this by overriding specific methods of the [ReportsControllerBase](/api/telerik.reporting.services.webapi.reportscontrollerbase) class in your `ReportsController`. This approach allows you to implement dynamic logic based on user context, report properties, or business rules.
+
+You can override the methods described in the following sections and customize different aspects of the AI-powered insights workflow.
+
+### CreateAIThread(string, string, ClientReportSource)
```

Comment:
From the explanations UpdateAIThread is called from base.CreateAIThread . While canceling the creation seem the perfect use case for CreateAIThread, the two other examples seem to fit better on UpdateUIThread, as we spare a cast for the developer - they get the AIThreadInfo directly as a parameter and can modify it. I would consider moving the examples 2 and 3 from CreateAIThread to the UpdateAIThread section, making them easier/shorter to implement - to achieve the desired goal. No need to show two ways to do one and the same thing, show me the recommended one only. If, on the contrary, there is a good reason for them to be in CreateAIThread, then explain it to me. Add what extra information or control I would have in CreateAIThread if there is such.